src/target/page/TargetPage.tsx

```diff
@@ -0,0 +1,12 @@
+import React, { useEffect, useState } from 'react';
+import apiRequest, { Target } from '../../common/utils/api-request';
+
+export default function TargetPage({ id }: { id: string }): JSX.Element {
+  const [target, setTarget] = useState<Target | null>(null);
+
+  useEffect(() => {
+    apiRequest.getTarget(id).then(setTarget);
```

Comment:
UseEffect's callbacks are synchronous to prevent race condition.
To circumvent this, it would have to be like this:
```js
useEffect(() => {
    const fetchTarget = async () => {
      const targetResponse = await api.getTarget(id);
      setTarget(targetResponse);
    };

    fetchTarget();
  }, [id]);
```

Reply:
Solution would be too complicated, I say let's go with regular Promise.then